hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/SnapshotDeletingService.java

```diff
@@ -564,6 +569,14 @@ public void submitRequest(OMRequest omRequest) {
     }
   }
 
+  public static boolean shouldIgnoreSnapshot(SnapshotInfo snapInfo,
+      SnapshotInfo.SnapshotStatus snapshotStatus,
+      boolean isSstFilteringServiceEnabled) {
+    return !snapshotStatus.equals(
+        SnapshotInfo.SnapshotStatus.SNAPSHOT_DELETED)
+        || (isSstFilteringServiceEnabled && !snapInfo.isSstFiltered());
```

Comment:
What is `(isSstFilteringServiceEnabled && !snapInfo.isSstFiltered());` check doing? It is saying that you will try to clean up a snapshot if SST filtering service is enabled but snapshot is not filtered.

Reply:
Clean up will only run if the SST Filtering service has been run on the snapshot.

Reply:
Yes, I get that but why do you need that check? If `Snapshot` is marked `DELETED`, we should be able to delete it.

Reply:
We could land in a race condition where the rocksDB will be open by SSTFilteringService and the OmSnapshotPurgeResponse could delete the checkpoint directory. It would be better to work with only closed rocksdb instances.

Reply:
Yes, but in that case, SFS will fail for that snapshot.

Created a jira: https://issues.apache.org/jira/browse/HDDS-9135 to revisit the snapshot deletion and filtering synchronization.